vividus/src/main/java/org/vividus/steps/VariablesSteps.java

```diff
@@ -60,7 +61,12 @@ public VariablesSteps(VariableContext variableContext, ISoftAssert softAssert,
             {
                 String readableCondition = EnumUtils.toHumanReadableForm(condition);
                 String description = "Checking if \"" + value1 + "\" is " + readableCondition + " \"" + value2 + "\"";
-                return softAssert.assertThat(description, value1, condition.getComparisonRule(value2));
+                boolean passed = softAssert.assertThat(description, value1, condition.getComparisonRule(value2));
+                if (!passed)
+                {
+                    diffAttachmentPublisher.publishDiff(value1, value2);
```

Comment:
I can't agree here, it might be useful in the case of invisible chars and other cases, because it highlights corner cases much better then plain old assertion